### PR TITLE
Do not add null checks for boxed statics

### PR DESCRIPTION
**Pull request by SingleAccretion**
While investigating #77773, I noticed we were adding null checks for `ref s_structStatic.Field`, which is not necessary.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
This should clean the Fuzzlyn pipeline from some number of #77773-related failures, obviously, it does not solve the underlying issue.

@dotnet/jit-contrib

**Review by jakobbotsch — APPROVED** [review]

**Comment by EgorBo** [comment]
Failures seem related? Apparently some other type of GT_IND_NONNULL node has GTF_EXCEPT

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -6566,7 +6566,7 @@ ExceptionSetFlags GenTree::OperExceptions(Compiler* comp)
                 return ExceptionSetFlags::NullReferenceException;
             }
 
-            return ExceptionSetFlags::None;
+            return ExceptionSetFlags::IndexOutOfRangeException;
```
Comment (jakobbotsch) [inline src/coreclr/jit/gentree.cpp:6569]: The case above also needs this I assume?

Comment (SingleAccretion) [reply src/coreclr/jit/gentree.cpp:6569]: Yes... it is getting late in the day.

Technically `ARR_INDEX` also needs it. Left it alone for now, we presumably expect to delete it.

**Comment by jakobbotsch** [comment]
> [Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=71253&view=results): a good amount of regressions due to the `ARR_ELEM` fix. We can take them here, or extract the `ARR_ELEM` change into a different PR.

I think it's fine to take it here. Thanks for looking at them.

**Pull request merged**